### PR TITLE
DDF-2378 Add 'internal' to registry packages that export service interfaces as well as the rest endpoint path

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Add 'internal' to registry packages that export service interfaces and to the rest endpoint path

Most all of the osgi services that were created with the experimental banner in the class header but should be updated to contain 'internal' in the package so that it is very clear it is not an external interface and can be changed. 
This should also be the changed for the registry rest endpoint. It is currently ../services/registries/...
and should be changed to ../services/internal/registries/...

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard @gordocanchola @vinamartin 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@shaundmorris

#### How should this be tested?
Build, ensure registry endpoints still function as expected

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by vinamartin** [comment]
Re-running bamboo, was seeing test failures.

**Comment by shaundmorris** [comment]
👍 What is our approach for the 2.9.x branch? Are we looking to mark those as internal as well?

**Comment by mcalcote** [comment]
I think so since this change is part of "feature completeness"

File: catalog/spatial/registry/registry-rest-endpoint/src/main/resources/templates/report.hbt [hunk 1]
```diff
@@ -82,7 +82,7 @@
             margin-left: 50px
         }
     </style>
-    <link href="/services/registries/assets/css/bootstrap-theme.min.css" rel="stylesheet" />
+    <link href="/services/internal/registries/assets/css/bootstrap-theme.min.css" rel="stylesheet" />
 </head>
```
Comment (ani6gup) [inline catalog/spatial/registry/registry-rest-endpoint/src/main/resources/templates/report.hbt]: this has to be modified in all the report handlebar templates `report, summary, organizations, and error` also there is a `report.css` class as a link too

File: catalog/spatial/registry/registry-rest-endpoint/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 2]
```diff
@@ -38,7 +38,7 @@
         <property name="registryReportMapBuilder" ref="registryReportMapBuilder"/>
     </bean>
 
-    <jaxrs:server id="registryRestEndpointService" address="/registries">
+    <jaxrs:server id="registryRestEndpointService" address="/internal/registries">
         <jaxrs:serviceBeans>
```
Comment (ani6gup) [inline catalog/spatial/registry/registry-rest-endpoint/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: you should modify the path used in `testRestEndpoint` for the itest `TestRegistry` (line 278)

File: catalog/spatial/registry/registry-rest-endpoint/src/main/resources/templates/report.hbt (hunk 1 above)
Comment (mcalcote) [inline catalog/spatial/registry/registry-rest-endpoint/src/main/resources/templates/report.hbt]: Thanks for the heads up!

File: catalog/spatial/registry/registry-rest-endpoint/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 2 above)
Comment (mcalcote) [inline catalog/spatial/registry/registry-rest-endpoint/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Will check it out :+1:

File: catalog/spatial/registry/registry-common/src/main/java/org/codice/ddf/registry/common/metacard/RegistryObjectMetacardType.java [hunk 3]
```diff
@@ -52,7 +52,7 @@
     public static final String DATA_END_DATE = "data-end-date";
 
     public static final String LINKS = "links";
-
+
```
Comment (mcalcote) [inline catalog/spatial/registry/registry-common/src/main/java/org/codice/ddf/registry/common/metacard/RegistryObjectMetacardType.java]: WHYY??? I banish ye to the depths!

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java [hunk 4]
```diff
@@ -197,6 +197,13 @@ public void updateRegistryEntry(Metacard updateMetacard) throws FederationAdminE
     public void updateRegistryEntry(Metacard updateMetacard, Set<String> destinations)
             throws FederationAdminException {
         validateRegistryMetacards(Collections.singletonList(updateMetacard));
+        String registryId = RegistryUtility.getRegistryId(updateMetacard);
+        if (registryId == null) {
+            return;
+        }
+
+        List<Metacard> existingMetacards =
+                this.getRegistryMetacardsByRegistryIds(Collections.singletonList(registryId));
```
Comment (mcalcote) [inline catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java]: NOPE!

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptionsTest.java [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+                id));
+        registryMetacard.setAttribute(new AttributeImpl(Metacard.MODIFIED,
+                new Date(new Date().getTime() + dateOffset)));
+        registryMetacard.setAttribute(new AttributeImpl(Metacard.TAGS,
+                Collections.singletonList(RegistryConstants.REGISTRY_TAG)));
+        registryMetacard.setAttribute(new AttributeImpl(Metacard.ID, TEST_METACARD_ID));
+        registryMetacard.setAttribute(new AttributeImpl(Metacard.METADATA, TEST_XML_STRING));
+        return registryMetacard;
+    }
+}
```
Comment (mcalcote) [inline catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptionsTest.java]: what the heck...

File: catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpointTest.java [hunk 6]
```diff
@@ -32,10 +32,10 @@
...
 
 import org.codice.ddf.parser.ParserException;
-import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
-import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
-import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
-import org.codice.ddf.registry.rest.endpoint.report.RegistryReportBuilder;
+import org.codice.ddf.registry.federationadmin.service.internal.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.internal.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.internal.RegistryPublicationService;
+import org.codice.ddf.registry.rest.endpoint.report.RegistryReportMapBuilder;
```
Comment (mcalcote) [inline catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpointTest.java]: get back

File: catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportMapBuilderTest.java [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+                JAXBElement.class,
+                getClass().getResourceAsStream(path));
+
+        if (jaxbRegistryObject != null) {
+            registryObject = jaxbRegistryObject.getValue();
+        }
+
+        return registryObject;
+    }
+}
```
Comment (mcalcote) [inline catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportMapBuilderTest.java]: Zombie Test Class

**Comment by adimka** [comment]
Successfully hero'd.

**Comment by pklinef** [comment]
:+1:

**Pull request merged**